contrib/juliac/juliac-buildscript.jl

```diff
@@ -27,6 +27,62 @@ if Base.JLOptions().trim != 0
     include(joinpath(@__DIR__, "juliac-trim-base.jl"))
 end
 
+const C_friendly_types = Base.IdSet{Any}([    # a few of these are redundant to make it easier to maintain
+    Int8, Int16, Int32, Int64, UInt8, UInt16, UInt32, UInt64, Float32, Float64, Bool,
+    Cvoid, Cint, Cshort, Clong, Cuint, Cushort, Culong, Cssize_t, Csize_t,
+    Cchar, Cwchar_t, Cstring, Cwstring,
+    RawFD,
+])
+
+function recursively_add_types!(types::Base.IdSet{DataType}, @nospecialize(T::DataType))
+    if T ∉ C_friendly_types
+        if T <: Ptr
+            return recursively_add_types!(types, T.parameters[1])
+        end
+        T.name.module === Core && error("invalid type for juliac: ", T) # exclude internals (they may change)
+        push!(types, T)
+    end
+    for list in (T.parameters, fieldtypes(T))
+        for S in list
+            recursively_add_types!(types, S)
+        end
+    end
+    return types
+end
+
+function write_logfile(io::IO)
```

Comment:
My KISS instinct would be to use a standard serialization format like JSON to print out the type information

Also do we need the `offset` info? No problem with adding it so I'm mostly asking for understanding - IIUC the offsets should be implied by the C ABI on a particular platform

Reply:
Totally happy to receive help! The more hands the better.

Reply:
Some changes at https://github.com/JuliaInterop/JuliaLibWrapping.jl/pull/2

it will need some more tweaks + new tests next week, but all tests are passing for now